pkg/k8sutil/create_test.go

```diff
@@ -287,3 +288,177 @@ func TestCreateOrUpdateNamespaceUpdateSuccess(t *testing.T) {
 		t.Fatalf("expected %q, got: %q", name, mockns.ObjectMeta.Labels[internal.NamespaceLabelKey])
 	}
 }
+
+func TestYAMLToUnstructured(t *testing.T) {
+	tests := []struct {
+		name    string
+		yamlObj []byte
+		want    *unstructured.Unstructured
+		wantErr bool
+	}{
+		{
+			name: "Valid_Kubernetes_object",
+			yamlObj: []byte(`apiVersion: v1
+kind: Namespace
+metadata:
+  name: test`),
+			want: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "v1",
+					"kind":       "Namespace",
+					"metadata":   map[string]interface{}{"name": "test"},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:    "Invalid_Kubernetes_object",
+			yamlObj: []byte(`foobar`),
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := YAMLToUnstructured(tt.yamlObj)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("got error: %v\nwantErr: %v", err, tt.wantErr)
+
+				return
+			}
+
+			if err != nil && tt.wantErr {
+				t.Logf("successfully failed with error: %v", err)
+
+				return
+			}
+
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("got: %#v\nwant: %#v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestSplitYAMLDocuments(t *testing.T) { //nolint:funlen
+	tests := []struct {
+		name    string
+		yamlObj string
+		want    []string
+		wantErr bool
+	}{
+		{
+			name: "valid_YAML_files",
+			yamlObj: `
+foo
+---
+bar`,
+			want:    []string{"foo\n", "bar\n"},
+			wantErr: false,
+		},
+		{
+			name: "valid_YAML_with_comments",
+			yamlObj: `# This is comment 1
+foo
+---
+# This is comment 2
+bar`,
+			want:    []string{"foo\n", "bar\n"},
+			wantErr: false,
+		},
+		{
+			name: "invalid_YAML",
+			yamlObj: "	foobar",
```

Comment:
How this is invalid YAML? https://yaml-online-parser.appspot.com/ and `yq` says it's valid.

Reply:
Oh wow, this is hard to spot. I'd use some more obvious example like:
```console
$ echo "foo: bar:" | yq
yq: Error running jq: ScannerError: mapping values are not allowed here
  in "<stdin>", line 1, column 9.
```

Reply:
Added one more test for invalid YAML.

- Tab char fails with error: `found character that cannot start any token`.
- `foo: bar:` fails with error: `mapping values are not allowed in this context`.

So kept both.